esphome/components/i2c/__init__.py

```
@@ -102,6 +126,8 @@ def _final_validate(config):
     full_config = fv.full_config.get()[CONF_I2C]
     if CORE.using_zephyr and len(full_config) > 1:
         raise cv.Invalid("Second i2c is not implemented on Zephyr yet")
+    if CORE.using_esp_idf and get_esp32_variant() in VALIDATE_LP_I2C:
+        config[CONF_LOW_POWER_MODE] = VALIDATE_LP_I2C[get_esp32_variant()](config)
```

I would rather throw an error then change the config to add lp mode

---

Thank you Jonathan, I will make the requested changes